dhall/src/Dhall/Binary.hs

```diff
@@ -426,8 +428,7 @@ instance ToTerm a => ToTerm (Expr s a) where
       where
         t₁  = encode t₀
         _T₁ = encode _T₀
-    encode (Note _ e) =
-        encode e
+    encode (Note (X absurd) _) = absurd
```

Comment:
I think it's a bit ugly that we have to include this alternative although the type proves that `Note`s are impossible. I think I'm in favor of #1120.

Reply:
You would still need to include that alternative even if #1120 were implemented.  The only difference is that you would write:

```haskell
encode (Note x _) = case x of {}
```

Reply:
Oh, that's sad. Not sure where I got that misconception from… :/